client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf, _n } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { ExternalLink, Button } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { recordEvent } from 'lib/tracks';
+
+/**
+ * Internal dependencies
+ */
+import '../style.scss';
+import DismissModal from '../dismiss-modal';
+import { getSetting } from '@woocommerce/wc-admin-settings';
+import withSelect from 'wc-api/with-select';
+
+const wcAdminAssetUrl = getSetting( 'wcAdminAssetUrl', '' );
+
+class ShippingBanner extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			showShippingBanner: true, // TODO: update to get state when closedForever is clicked
+			isDismissModalOpen: false,
+		};
+	}
+
+	componentDidMount() {
+		this.trackBannerEvent( 'shipping_banner_show' );
+	}
+
+	closeDismissModal = () => {
+		this.setState( { isDismissModalOpen: false } );
+		this.trackBannerEvent(
+			'shipping_banner_dismiss_modal_close_button_click'
+		);
+	};
+
+	openDismissModal = () => {
+		this.setState( { isDismissModalOpen: true } );
+		this.trackBannerEvent( 'shipping_banner_dimiss_click' );
+	};
+
+	hideBanner = () => {
+		this.setState( { showShippingBanner: false } );
+	};
+
+	createShippingLabelClicked = () => {
+		// TODO: install and activate WCS
+		// TODO: open WCS modal
+		this.trackBannerEvent( 'shipping_banner_create_label_click' );
+	};
+
+	learnMoreClicked = () => {
+		this.trackBannerEvent( 'shipping_banner_learn_more_click' );
+	};
+
+	trackBannerEvent = ( eventName ) => {
+		recordEvent( eventName, {
+			jetpack_installed: this.activePlugins.includes( 'jetpack' ),
+			jetpack_connected: this.isJetpackConnected,
```

Comment:
I think you've connected the component successfully, this just needs to be `this.props.isJetpackConnected`.

You can also destructure these above to make things a bit easier to grok:
`const { activePlugins, isJetpackConnected } = this.props`